src/runtime/vm/vm.cc

```diff
@@ -921,7 +923,7 @@ void VirtualMachine::RunLoop() {
         CHECK(tuple != nullptr)
             << "Object is not data type object, register " << instr.object << ", Object tag "
             << object->type_index();
-        auto field = tuple->fields[instr.field_index];
+        auto field = (*tuple)[instr.field_index];
```

Comment:
Let us also overload operator[] in the ADT container class, so we can do tuple[instr.field_index]

Reply:
done.